spec/models/playbook_spec.rb

```diff
@@ -26,6 +26,59 @@
     expect(playbook).to be_valid
   end
 
+  describe '#archive!' do
+    before do
+      allow(Time).to receive(:now)
+                 .and_return(Time.parse("2021-02-23 19:21:00"))
+      playbook.archive! 
+    end
+
+    it 'sets the archived date' do
+      expect(playbook.archived_at).not_to be_nil
+    end
+
+    context 'with created playbook' do
+      let(:playbook) { create :playbook }
+
+      it 'retains the archived_at' do
+        expect(playbook.reload.archived_at).not_to be_nil
+      end
+    end
+
+    context 'when playbook has already been archived' do
+      let(:date) { Time.parse('2021-02-20 19:21:00') }
+      let(:playbook) { build :playbook, archived_at: date }
+
+      it "doesn't overwrite existing archived_at" do
+        expect(playbook.archived_at).to eq date
+      end
+    end
+  end
+
+  describe '.unarchived' do
+    subject { Playbook.unarchived }
+    let(:date) { Time.parse('2021-02-20 19:21:00') }
+    let!(:archived) { create :playbook, archived_at: date }
+    let!(:unarchived) { create :playbook }
+
+    it 'only unarchived values' do
+      is_expected.to include unarchived
+      is_expected.not_to include archived
+    end
+  end
+
+  describe '.all' do
+    subject { Playbook.all }
+    let(:date) { Time.parse('2021-02-20 19:21:00') }
+    let!(:archived) { create :playbook, archived_at: date }
+    let!(:unarchived) { create :playbook }
+
+    it 'only unarchived values' do
+      is_expected.to include unarchived
+      is_expected.not_to include archived
+    end
+  end
+
```

Comment:
Chael's specs - I liked this exercise, thank you!
![](https://media.giphy.com/media/26tPplGWjN0xLybiU/giphy.gif)